### PR TITLE
Auto-Focus Group on Creation

### PR DESCRIPTION
**Pull request by m-peeler**
When a new group is created, it is automatically focused in the viewer. This addresses and closes https://github.com/JabRef/jabref/issues/11449.

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: CHANGELOG.md
Comment (koppor) [inline CHANGELOG.md:32]:
```suggestion
- We made new groups automatically focus upon creation. [#11449](https://github.com/JabRef/jabref/issues/11449)
```

**Review by koppor — CHANGES_REQUESTED** [review]
Besides minor comments, LGTM :+1:. The changes need a second review and thus this PR awaits a second review.

**Comment by koppor** [comment]
Due to the discussion at https://github.com/JabRef/jabref/issues/11449 postponed to 6.0-alpha

File: src/main/java/org/jabref/gui/StateManager.java
Comment (LoayGhreeb) [inline src/main/java/org/jabref/gui/StateManager.java:143]: This is incorrect. Do not clear the entire map, as this will result in losing the selected groups when switching between open libraries.
```suggestion
```

**Review by LoayGhreeb — APPROVED** [review]

**Comment by koppor** [comment]
@m-peeler I needed to merge `main` and update `CHANGELOG.md`, because we had a release meanwhile. And due to the discussion at the issue, I did not dare to distribute that functionality in a release.

**Review by koppor — APPROVED** [review]

**Pull request merged**
